src/rgb_led.rs

```diff
@@ -0,0 +1,230 @@
+// Modified from https://github.com/fkohlgrueber/esp32c3-idf-led-example
```

Comment:
Understand it's modified from another source. Can we still add a couple comments to describe program usage?

Reply:
The example file contains the program usage.
The file structure looks looks complicated, but the only things that are public are the Led struct (and builder), its methods `set_color` and `fade_to`, and their associated error. Each of these should be self-explanatory.